### PR TITLE
[rrd4j] Improve performance

### PR DESCRIPTION
**Pull request by wborn**
* Cache item to improve performance
* Also return DecimalType for QuantityType values

Related to #8928
Reintroduces #8809

### CODE REVIEW

**Review by J-N-K — APPROVED** [review]

**Comment by J-N-K** [comment]
@kaikreuzer Please wait with the M2 build until this is merged. I have triggered rebuild because the previous build faild while resolving the tado-library.

File: bundles/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jPersistenceService.java [hunk 1]
```diff
@@ -230,73 +230,83 @@ public void store(Item item) {
...
         String itemName = filter.getItemName();
+
         RrdDb db = getDB(itemName);
-        if (db != null) {
-            ConsolFun consolidationFunction = getConsolidationFunction(db);
-            long start = 0L;
-            long end = filter.getEndDate() == null ? System.currentTimeMillis() / 1000
-                    : filter.getEndDate().toInstant().getEpochSecond();
+        if (db == null) {
+            logger.warn("Could not find item '{}' in rrd4j database", itemName);
```
Comment (kaikreuzer) [inline bundles/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jPersistenceService.java:236]: Afaics, we didn't have this warning before - and I'd think that there might be legit situations for a query even if no db file is there (yet).

Comment (wborn) [reply bundles/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jPersistenceService.java:236]: Yes it was meant to be debug level. I'll change that.

Comment (wborn) [reply bundles/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jPersistenceService.java]: I only reduced the amount of of nesting/indentation with this quick return statement. That makes the code a bit more readable. Moving logic to other methods also helps with that.

Comment (kaikreuzer) [reply bundles/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jPersistenceService.java]: Yes, I saw and understood that - although it made it trickier to review and see the parts that are the real change in it.

**Review by kaikreuzer — APPROVED** [review]
Jenkins failed on feed tests, I guess we can ignore that.

**Pull request merged**